server/cpho/views/submission.py

```diff
@@ -113,3 +134,25 @@ def get_context_data(self, *args, **kwargs):
         return {
             **super().get_context_data(*args, **kwargs),
         }
+
+    def get(self, request, *args, **kwargs):
+        context = self.get_context_data(**kwargs)
+        if not (
+            test_rule("is_admin", request.user)
+            or (
+                test_rule("is_program", request.user)
+                and test_rule(
+                    "can_submit_as_program", request.user, self.indicator
+                )
+            )
+            or (
+                test_rule("is_hso", request.user)
+                and test_rule(
+                    "can_submit_as_hso", request.user, self.indicator
+                )
+            )
+        ):
```

Comment:
Views combining several rules is an anti-pattern, usually it means this expression should be its own rule